addons/mail/wizard/mail_activity_schedule_wizard.py

```diff
@@ -0,0 +1,315 @@
+                for error in errors:
+                    error_display += '<li>%s</li>' % error
+                error_display += '</ul>'
+            else:
+                error_display = False
+            wizard.error = error_display
+
+    @api.depends('error')
+    def _compute_has_error(self):
+        for wizard in self:
+            self.has_error = wizard.error and wizard.error != '<p><br></p>'
+
+    @api.depends('res_ids')
+    def _compute_is_batch_mode(self):
+        for wizard in self:
+            self.is_batch_mode = len(wizard._get_converted_res_ids(wizard.res_ids)) > 1
+
+    @api.onchange('activity_type_id')
+    def _onchange_activity_type_id(self):
+        self.env['mail.activity']._apply_activity_type_defaults(self.activity_type_id, self)
+
+    def _get_activities_to_schedule(self):
+        return self.plan_id.template_ids
+
+    @api.model
+    def _get_converted_res_ids(self, res_ids_str):
+        """ Convert list of ids separated by comma into list of id number. """
+        return [int(res_id_str) for res_id_str in res_ids_str.split(',')]
+
+    def _get_records(self):
+        return self.env[self.res_model].browse(self._get_converted_res_ids(self.res_ids))
+
+    @api.model
+    def _get_record_for_scheduling(self, record, responsible):
+        """ Get the record on which the activity will be linked when launching a plan.
+
+        :param <res.user> responsible: responsible
+
+        The base implementation returns the record itself which is the common case.
+        This method is meant to be overriden in other modules to return a different
+        record if needed. That method has been introduced for the hr module where we
+        needed to link the activity to another record for security reason.
+        """
+        return record
+
+    def action_schedule_plan(self):
+        records = self._get_records()
+        for record in records:
+            body = _('The plan "%s" has been started', self.plan_id.name)
+            activities = set()
+            for activity in self._get_activities_to_schedule():
+                if activity.responsible == 'on_demand':
+                    responsible = self.on_demand_user_id
+                else:
+                    responsible = activity._get_responsible_id(self.on_demand_user_id, record)['responsible']
+                date_deadline = self.env['mail.activity']._calculate_date_deadline(
+                    activity.activity_type_id) if not self.date_plan_deadline else self.date_plan_deadline
+                self._get_record_for_scheduling(record, responsible).activity_schedule(
+                    activity_type_id=activity.activity_type_id.id,
+                    summary=activity.summary,
+                    note=activity.note,
+                    user_id=responsible.id,
+                    date_deadline=date_deadline
+                )
+                activity = _('%(activity)s, assigned to %(name)s, due on the %(deadline)s',
```

Comment:
Don't rewrite on 'activity' variable of the loop
```suggestion
                activity_description = _('%(activity)s, assigned to %(name)s, due on the %(deadline)s',
```